pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -783,15 +784,21 @@ const (
 // ApplicationSetApplicationStatus contains details about each Application managed by the ApplicationSet
 type ApplicationSetApplicationStatus struct {
 	// Application contains the name of the Application resource
-	Application string `json:"application" protobuf:"bytes,1,opt,name=application"`
-	// LastTransitionTime is the time the status was last updated
-	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,2,opt,name=lastTransitionTime"`
-	// Message contains human-readable message indicating details about the status
-	Message string `json:"message" protobuf:"bytes,3,opt,name=message"`
-	// Status contains the AppSet's perceived status of the managed Application resource: (Waiting, Pending, Progressing, Healthy)
-	Status string `json:"status" protobuf:"bytes,4,opt,name=status"`
-	// Step tracks which step this Application should be updated in
-	Step string `json:"step" protobuf:"bytes,5,opt,name=step"`
+	Application string       `json:"application" protobuf:"bytes,1,opt,name=application"`
+	UID         types.UID    `json:"uid" protobuf:"bytes,2,opt,name=uid,casttype=k8s.io/kubernetes/pkg/types.UID"`
+	CreatedAt   *metav1.Time `json:"createdAt,omitempty" protobuf:"bytes,3,opt,name=createdAt"`
+	Health          HealthStatus           `json:"health,omitempty" protobuf:"bytes,4,opt,name=health"`
+
+  // Progressive Sync fields
+	// ProgressiveSyncLastTransitionTime is the time the status was last updated
+	ProgressiveSyncLastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,5,opt,name=lastTransitionTime"`
+	// ProgressiveSyncMessage contains human-readable message indicating details about the status of the progressive sync
+	ProgressiveSyncMessage string `json:"message" protobuf:"bytes,6,opt,name=message"`
+	// ProgressiveSyncStatus contains the AppSet's perceived status of the managed Application resource: (Waiting, Pending, Progressing, Healthy)
+  // this is NOT the same as the Application's status
+	ProgressiveSyncStatus string `json:"status" protobuf:"bytes,7,opt,name=status"`
+	// ProgressiveSyncStep tracks which step the Applications should be updated in
+	ProgressiveSyncStep string `json:"step" protobuf:"bytes,8,opt,name=step"`
```

Comment:
To avoid a breaking gRPC change, keep the number in the `protobuf` tag for each of these the same and add the new fields after these. It might read a bit wonky, but worth it to avoid the breaking API change.

Reply:
Ah ok thanks for the info! I was wondering if it had an impact

Reply:
Done!